### ISSUE TITLE
Inconsistent ENV usage across languages is problematic

### ISSUE THREAD

**Issue by bootstraponline** [saucelabs/sauce_whisk#28]
sauce_whisk

> ENV['SAUCE_USERNAME']   ||= 'test_user'
> ENV['SAUCE_ACCESS_KEY'] ||= 'test_key'

[sauce-java](https://github.com/saucelabs/sauce-java/blob/81d3ff0619c3216726e84d5d573a587c3b6ee095/common/src/main/java/com/saucelabs/common/SauceOnDemandAuthentication.java#L30)

> private static final String SAUCE_USER_NAME = "SAUCE_USER_NAME";
>     private static final String SAUCE_API_KEY = "SAUCE_API_KEY";

I think Sauce should have a standard naming convention for the ENV vars instead of different values for each project.

**Comment by bootstraponline** [comment]
The [sauce java client ended up supporting](https://github.com/saucelabs/sauce-java/blob/df707de514819497963da831c35a6b20b881476d/common/src/main/java/com/saucelabs/common/SauceOnDemandAuthentication.java#L33):

- SAUCE_USER_NAME
- SAUCE_USERNAME
- SAUCE_API_KEY
- SAUCE_ACCESS_KEY

There's still a great deal of inconsistency depending on the clients (most use only two from the above list). The java client is the only one that supports all 4. It'd be good to know what ones are "official."

**Comment by jlipps** [comment]
i don't think Sauce can have any stance on 'official' names for env vars. but i agree that the various clients should endeavor to use the same thing to avoid surprise.

**Comment by bootstraponline** [comment]
Sauce could say here's what we're supporting consistently across the clients instead of using different names in the docs and each language. That's what I meant by 'official.' People run into the [same issue](https://discuss.appium.io/t/appium-lib-and-sauce/4092) with the appium clients. I've settled on `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` since that tends to be used more frequently. If people are using the other names then it's going to break. Ultimately I think the official sauce clients should support all 4 since it's not the users fault there's inconsistency in naming.

**Comment by jlipps** [comment]
fair enough. i think the sauce docs should always use 1 style as an example, and have all the clients support all 4 styles.

**Comment by bootstraponline** [comment]
sounds good to me :+1:

**Comment by DylanLacey** [comment]
I agree it'd be good to at least standardise on what we do it in docs;  Personally I like `API_KEY` but the term `ACCESS_KEY` is more widely used.
